pandas/tseries/period.py

```diff
@@ -306,6 +307,23 @@ def __contains__(self, key):
             return False
         return key.ordinal in self._engine
 
+    def __array_wrap__(self, result, context=None):
+        """
+        Gets called after a ufunc. Needs additional handling as
+        PeriodIndex stores internal data as int dtype
+        """
+        if isinstance(context, tuple) and len(context) > 0:
+            func = context[0]
+            if (func is np.add):
+                return self._add_delta(context[1][1])
+            elif (func is np.subtract):
+                return self._add_delta(-context[1][1])
+            elif isinstance(func, np.ufunc):
+                if 'M->M' not in func.types:
+                    msg = "ufunc '{0}' not supported for the PeriodIndex"
+                    raise ValueError(msg.format(func.__name__))
```

Comment:
@shoyer you know any better way to override the ufunc calling scheme? (e.g. as ``__numpy_ufunc__`` doesn't exist yet). or is this best way?